fastn-update/src/lib.rs

```diff
@@ -2,60 +2,91 @@ extern crate self as fastn_update;
 
 mod utils;
 
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Failed to resolve manifest.json for package '{package}'")]
+    ManifestResolution {
+        package: String,
+        #[source]
+        error: fastn_core::Error,
+    },
+    #[error("Failed to download and unpack archive for package '{package}'")]
+    ArchiveDownload {
+        package: String,
+        #[source]
+        error: fastn_core::Error,
+    },
+    #[error("Failed to resolve dependency '{package}'")]
+    DependencyResolution {
+        package: String,
+        #[source]
+        error: fastn_core::Error,
+    },
+}
```

Comment:
Don't use this pattern for new code. Create a unique error enum for every function called with precise error variants.